.github/workflows/ci.yml

```
@@ -0,0 +1,45 @@
+name: CI - Run Tests
+
+on:
+  push:
+    branches:
+      - main
+      - stage
+  pull_request:
+    branches:
+      - main
+      - stage
```

this is simpler and cover all your usage:

```suggestion
on:
  push:
    branches:
      - "**"
```